Support user defined delimiter when parsing list

rafiw opened the pull request:
Currently user can only parse list with spaces
E.g app -o 1 2 3
now use can define his own one char delimiter e.g comma

harrywong commented on tests/AppTest.cpp:
```diff
@@ -339,7 +339,7 @@ TEST_F(TApp, doubleVectorFunction) {
     args = {"--val", "5", "--val", "6", "--val", "7"};
     run();
-    EXPECT_EQ(res.size(), 3);
+    EXPECT_EQ(res.size(), (size_t)3);
```
I think `EXPECT_EQ(res.size(), 3u);` is better here :)

henryiii replied:
That might be a good idea generally, but for now, this is consistent.

henryiii replied:
Okay, this can be changed now. I've converted to using `#u` everywhere in testing. Thanks!

henryiii committed:
Add support for delimiter when parsing vector

This commit allows parsing any char separated list given by the user.
E.g app -a 1,2,3
std::vector<int> params;
app.add_option("-a", params, "Parse the params", ',');

Signed-off-by: Rafi Wiener <rafiw@mellanox.com>

add tests for delimiter parsing

Signed-off-by: Rafi Wiener <rafiw@mellanox.com>

Fixing style, adding docker version of clang-format
